pxd.c

```diff
@@ -1621,6 +1646,39 @@ ssize_t pxd_remove(struct fuse_conn *fc, struct pxd_remove_out *remove)
 	get_device(&pxd_dev->dev);
 	prepare_to_wait(&pxd_dev->remove_wait, &wait, TASK_INTERRUPTIBLE);
 	spin_unlock(&pxd_dev->lock);
+
+	/// perform all below actions on the kernel object if device got exported.
+	if (pxd_dev->exported) {
```

Comment:
i do not quite follow here, why this needs to be done? This entire logic seems is already part of pxd_finish_remove that is scheduled through a workqueue above. Did i missing anything?

Reply:
merge issue, fixed.